### PR TITLE
Pull request 20231213

### PR DESCRIPTION
**Pull request by hayatehirai**
Most changes are for sorting the compounds before finding minimal seed sets. The modified code enables you to:
- Move target compounds to the top of the list
- Sort compounds by exact mass in descending order 
- Exchange compounds while maintaining the descending trend

### CODE REVIEW

File: src/BioXP.jl [hunk 1]
```diff
@@ -218,7 +218,7 @@ end
 """
 Return indices of seeds within the compounds vector. TEST
 """
-function seed_indicies(sids::IDs, cids::IDs)
+function seed_indices(sids::IDs, cids::IDs)
```
Comment (hbsmith) [inline src/BioXP.jl:221]: nice spelling catch

File: src/randomize.jl [hunk 2]
```diff
@@ -47,13 +55,18 @@ Swap random tuples within `tuples`.
...
+    tuples::Vector{<:Tuple{String,Any}},
     beta::Real,
     n_swaps::Int,
     rng::AbstractRNG=Random.default_rng()) #where {R <: Real}
 
     for _ in 1:n_swaps
-        i,j = rand(1:length(tuples),2) ## 2 random indices 
+        i,j = rand(1:length(tuples),2) ## two random indices
+        if i > j
+            i,j = j,i
```
Comment (hbsmith) [inline src/randomize.jl:66]: Doesn't this make it so that the `elseif` in the `getflipprobability` will never get triggered? Is that good or bad?

**Review by hbsmith — COMMENTED** [review]
Looks good to me besides my one comment

File: src/randomize.jl (hunk 2 above)
Comment (hayatehirai) [reply src/randomize.jl:66]: Thank you for checking it. This is to reduce the number of exchanges of compounds.
If the `i>j` cases are allowed, `diff = ti[2] - tj[2]` tends to be negative (Positions of compound i and compound j are like [c1, ... , cj, ... , ci, ... , cn]). As a result, compounds are much more likely to be flipped.
The attached figure shows the positions of compounds after random perturbation and their mass (a) when the `i>j` cases are allowed and (b) when not allowed.
I thought it's good to keep the trend of the initial order of compounds as figure b, so index j is to be larger than i in the modified code. `elseif` in the `getflipprobability` happens when compound i is heavier than compound j.
order_of_compound_after_perturbation

Comment (hbsmith) [reply src/randomize.jl:66]: got it! thanks for explaining/reminding me. everything looks good then

File: src/randomize.jl [hunk 3]
```diff
@@ -21,14 +21,21 @@ Always shuffles the order of the zero-mass compounds.
 function sort_biosystem_compounds(
     compound_structs::Compounds,
     biosystem_compounds::IDs,
+    tids::IDs,
     sortkey::Symbol=:exact_mass,
     zero_mass_behavior::String="end")
```
Comment (hbsmith) [inline src/randomize.jl:26]: We might want to just get rid of this keyword argument since it's not doing anything anymore

Comment (hayatehirai) [reply src/randomize.jl]: Sure, I deleted it

**Review by hbsmith — APPROVED** [review]
looks good!

**Pull request merged**